### DISCUSSION TITLE
Need Help Adding Content Script to Vite-Based extension

### DISCUSSION THREAD

**Discussion by akash-singh8** [open-sauced-craftwork/browser-extension#24] [Q&A] [answered]
#### I would appreciate any guidance or suggestions from the community on how to successfully incorporate this content script into my extension. Thank you in advance for your support.

**Comment by diivi** [comment]
Some points I'd recommend you to check: 
1. You are checking the correct console while logging "hello". Check both the main window's DevTools and the extension's DevTools.
2. You have the correct pattern matching set up in `manifest.json` maybe if you can provide that here and what you tried, it will be easier for us to debug.

**Reply by akash-singh8** [reply]
I've gone through this doc [https://crxjs.dev/vite-plugin/getting-started/vanilla-js/add-content-script](url)

**Reply by heysujal** [reply]
@akash-singh8 URL is not working. As @diivi said you should ensure that you are checking the correct console. You should check the chrome developer tools console. Also, you should right-click on the extension and click inspect and go to the console to check if your statement is being logged or not.

Provide any error message you are seeing or other things where you are stuck

**Reply by akash-singh8** [reply]
[Docs](https://crxjs.dev/vite-plugin/getting-started/vanilla-js/add-content-script)
Look the whole purpose of the content script is to execute in the context of each webpage. So there's no way for it to be executed in DevTools.

**Comment by nickytonline** [comment]
Whatever way you build your content script doesn’t matter. What’s important is that your [manifest includes a reference to the content script](https://developer.chrome.com/docs/extensions/mv3/content_scripts/#static-declarative).

**Reply by akash-singh8** [reply]
Yeah, I've done that as well.

**Comment by akash-singh8** [comment]
## It's fixed 

![image](https://user-images.githubusercontent.com/85285176/225709934-bb9fd6b4-f309-446a-9851-16f7ad761e00.png)

**Reply by heysujal** [reply]
@akash-singh8 Where is your console statement being printed now?
  is your extension working without the need of clicking on the icon or do you still need to click on it?

**Reply by akash-singh8** [reply]
Yeah, it is. @heysujal

**Comment by harshal015** [comment, marked as answer]
I encountered a similar issue before, I would suggest verifying that the file path is correct. Also, if your script file is located within a separate directory, it would be advisable to keep it to the same hierarchy as the manifest file.

**Reply by heysujal** [reply]
does your script work without clicking on the extension icon ?
